---
title: "Estimation of Genomic Breeding Values"
author: "Peter Sørensen,....."
date: "`r Sys.Date()`"
output:
  pdf_document:
    number_sections: yes
    includes:
      in_header: preamble.tex
  html_document:
    number_sections: yes
    includes:
      in_header: mathjax_header.html
  word_document: default
biblio-style: apalike
link-citations: yes
bibliography: lbgfs2021.bib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(19)
```



## Learning objective:  {-}    
This section introduces the basic concepts of estimating breeding values such as: 

   * basic principle behind estimating genomic breeding values
   * accuracy of estimated genomic breeding values
   * use of genomic relationships for estimating breeding values
   * different methods, data sources and experimental designs for estimating genomic breeding values


# Introduction
A new technology called __genomic selection__ has revolutionized animal and plant breeding. Genomic selection refers to selection decisions based on genomic breeding values (GEBV). We have previously learned how breeding values can be estimated from pedigree and phenotypic information (estimated breeding value, EBV). Genome-wide DNA markers can replace or supplement pedigree information for this purpose. The first ideas of genomic prediction were presented by (Meuwissen2001a). They showed that information from genotypes of very many loci evenly spread over the complete genome can successfully be used for the purposes of livestock breeding. Because the information of the genotypes is spread over the complete genome it is often referred to as __genomic information__ and from the use of this information for selection purposes the term of __genomic selection__ was invented. The early results on GS were not considered until the paper by (Schaeffer2006) showed that in a cattle breeding program the introduction of GS could lead to savings in about 90% of the total costs, provided that the accuracies computed by (Meuwissen2001a) can really be achieved. After the publication of (Schaeffer2006) many animal and plant breeding organisation started to introduce procedures of GS. 


## Basic principles for estimating genomic breeding values (GEBV)
In recent years much attention has been given to 
genomic information due to the dramatic development in genotyping technologies. 
Today dense 
genetic maps are available for most of the most important livestock species (Table 1). It is, 
however, still lacking for several species, but they circumvent this by using a so-called RAD-sequencing 
(Restriction site associated DNA sequencing) which enable dividing the entire genome into smaller 
segments just like if a genetic map had been available. Ultimately the entire DNA sequence may be 
genotyped. This is possible, but still very expensive, so only a few founder individual have been fully 
sequenced (mostly bulls, but also some horses and dogs). Lower resolution maps are also used and 
especially in cattle to save costs (e.g. females). 

Table 1. Number of markers in currently available SNP chips (March, 2013)
\begin{center} 
\begin{tabular}{|c|c|c|}
  \hline
  Species & No. SNPs (in thousands) & Genome size (x$10^9$) \\
  \hline
  Cattle & 778 & 2,67 \\
  \hline
  Pig & 64 & 2,81  \\
  \hline
  Chicken & 581 & 1,05 \\ 
  \hline
  Horse & 70 & 2,47  \\
  \hline
  Sheep & 54 & 2,62  \\
  \hline
  Dog & 170 & 2,41  \\
  \hline
\end{tabular}
\end{center}

The genetic maps are based on DNA markers in the form of single nucleotide polymorphisms 
(SNP) and they enable us to divide the entire genome into thousands of relatively small 
chromosome segments. 
 
## Genetic markers
The single location in the genome that are considered in GS are called __markers__. When looking at the complete set of markers consisting the genomic information in a population, the so-called __Single Nucleotide Polymorphisms__ (SNP) have been shown to be the most useful types of markers. These SNP correspond to differences of single bases at a given position in the genome. Based on empirical analyses of very many SNP-loci, almost all SNP just take two different states. Furthermore it is important that these SNPs are more or less evenly spread over the complete genome. Some SNPs are in coding regions and some my be placed in regions of unknown functionality. Figure \@ref(fig:snpdistribution) shows the distribution of SNP over the genome. 

```{r snpdistribution, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", fig.cap="Distribution of SNP-Loci Across A Genome"}
#rmddochelper::use_odg_graphic(ps_path = "odg/snpdistribution.odg")
knitr::include_graphics(path = "odg/snpdistribution.png")
```

## Quantitative Trait Loci
The loci that are relevant for a quantitative traits are called __Quantitative Trait Loci__ (QTL). Any given SNP-Marker can only be informative for a given QTL, if a certain __linkage disequilibrium__ between the QTL and the marker locus exists. The idea behind this linkage disequilibrium is that a certain positive QTL-allele evolved in a certain genetic neighborhood of a number of SNP loci. As a result of that the positive QTL-allele is very often inherited with the same SNP-allele. Over the generations, recombination between the QTL and the neighboring SNP-loci can happen and thereby weaken the association between the positive QTL-allele and the given SNP-allele. This recombination effect is smaller when the QTL and the SNP-loci are physically closer together on the chromosome. The non-random association between QTL and SNP-markers is called linkage disequilibrium.

The marker locus is called $M$ and the QTL is called $Q$, then the LD can be measured by 

\begin{align}
D &= p(M_1Q_1) * p(M_2Q_2) - p(M_1Q_2) * p(M_2Q_1)
(\#eq:linkagediseq)
\end{align}

where $p(M_xQ_y)$ corresponds to the frequency of the combination of marker allele $M_x$ and QTL allele $Q_y$. Very often the LD measure shown in \@ref(eq:linkagediseq) is re-scaled to the interval between $0$ and $1$ which leads to 

\begin{align}
r^2 &= \frac{D^2}{p(M_1)*p(M_2) * p(Q_1) * p(Q_2)}
(\#eq:linkagediseqrescaled)
\end{align}

In \@ref(eq:linkagediseqrescaled) $r^2$ describes the proportion of the variance at the QTL which is explained by the marker $M$. Hence the LD must be high such that the marker can explain a large part of the variance at the QTL. For the length of most livestock species, about $50'000$ SNP markers are required to get a sufficient coverage of the complete genome.

## Basic principles for estimating genomic breeding values (GEBV)
The GEBV are calculated as the sum of the effects of dense genetic markers across the entire genome, thereby potentially capturing all the quantitative trait loci (QTL) that contribute to variation in a trait. The QTL effects inferred from individual single nucleotide polymorphism (SNP) markers, are first estimated in a large reference population with phenotypic information. In subsequent generations, only marker information is required to calculate GEBV. 
Breeding values are conceptually simple to calculate from marker information. First, the entire 
genome is divided into small chromosome segments by dense markers. Second, the additive effects 
of each chromosome segment are estimated simultaneously. Finally, the genomic EBV equals the 
sum of all chromosome segment effects. The chromosome segment effects are estimated for a group 
of animals (i.e. a reference population). For any remaining animal, only a blood or tissue sample is 
needed to determine its genome-wide (or genomic) EBV (Figure 1). For breeding purposes, it is 
desirable that the EBV can be estimated accurately early in the animals life. 
The effect of each of these small chromosome segments can be estimated if we have phenotypes 
and genotypes from a number of animals. With sufficiently dense marker maps, the chromosome 
segment effects apply to all animals in the population in which they were estimated, because 
markers are in linkage disequilibrium with the causal gene that they bracket. 


## A linear mixed model for estimating marker effects (SNP-BLUP) {#linear-model-pgbv}
A linear model to estimate SNP-effects based on the data from the reference population in the two-step procedure can be defined as follows

\begin{equation}
y = X \beta +  Mg + e
(\#eq:linearmodelgpbv)
\end{equation}

\begin{tabular}{lll}
where  &  $m$  &  number of SNP markers \\
       &  $y$  &  vector of observations \\
       &  $\beta$  &  vector of fixed effects \\
       &  $X$      &  design matrix linking fixed effects to observations \\
       &  $g$      &  random genetic effect of SNP-genotypes \\
       &  $M$      &  design matrix linking SNP-genotype effects to observations \\
       &  $e$      &  vector of random residuals
\end{tabular}       

The observations $y$ used in \@ref(eq:linearmodelgpbv) are in most evaluations not phenotypes but traditionally predicted breeding values with an accuracy above a certain threshold. As a consequence of that the variance-covariance matrix ($R$) of the residuals $e$ is not just an identity matrix ($I$) times a residual variance component ($\sigma_e^2$) but $R$ is a diagonal matrix with elements $(R)_{ii} = {1\over B_{m}} - 1$ where $B_m$ is the accuracy of the traditionally predicted breeding value from an animal from the reference population, corrected for the parental contributions. In effect, $B_m$ corresponds to the accuracy of the mendelian sampling term.

The mixed-model equations resulting from models given in \@ref(eq:linearmodelgpbv) have the following structure

\begin{equation}
\left[
  \begin{array}{lr}
  X^TR^{-1}X  &  X^TR^{-1}M  \\
  M^TR^{-1}X  &  M^TR^{-1}M + I * \lambda
  \end{array}
\right]
\left[
  \begin{array}{c}
  \hat{\beta}\\
  \hat{g}
  \end{array}
\right]
=
\left[
  \begin{array}{c}
  X^TR^{-1}y \\
  M^TR^{-1}y
  \end{array}
\right]
(\#eq:mmegpbv)
\end{equation}

where 

\begin{equation}
\lambda = \frac{\sigma_e^2}{\sigma_a^2} \sum_{i=1}^m 2*p_i*(1-p_i)
(\#eq:lambdammegpbv)
\end{equation}

In \@ref(eq:lambdammegpbv) $\sigma_a^2$ is the total genetic variance and $p_i$ is the frequency of the SNP-allele that is associated with the positive QTL-allele.

The solutions for $\hat{g}$ from \@ref(eq:mmegpbv) correspond to the SNP-genotype effects. The predicted breeding value $\hat{a}$ for any selection candidate with genomic information is then computed as

\begin{equation}
\hat{a} = \sum_{i=1}^m M_i \hat{g}_i
(\#eq:genomicpbv)
\end{equation}

where $M_i$ corresponds to the vector of SNP-genotypes of the selection candidate.
This model is sometimes referred to the SNP-BLUP model. 


### Matrix $M$
The elements in matrix $M$ can be encoded in different ways. The results from the genotyping laboratory sends a code representing the nucleotide that can be found at a given position. For the use in the linear model we have to use a different encoding. Let us assume that at a given SNP-position, the bases $G$ or $C$ are observed and $G$ corresponds to the allele with the positive effect on our trait of interest. Based on the two observed alleles, the possible genotypes are $GG$, $GC$ or $CC$. One possible code for this SNP in the matrix $M$ might be the number of $G$-Alleles which corresponds to $2$, $1$ and $0$. Alternatively, it is also possible to use the codes $1$, $0$ and $-1$ instead which corresponds to the factors with which $a$ is multiplied to get the genotypic values in the single locus model.



## A linear mixed model for estimating genomic values (GBLUP) {#gblup}
The term `GBLUP` stands for genomic BLUP and is the most widely used single-step procedure. In GBLUP genomic breed values are directly predicted without the prediction of marker effects. This can be done by including the genomic breeding values $u$ which corresponds to the sum of all SNP-allele effects directly as a random effect in the model. 

\begin{equation}
y = X \beta + W u + e
(\#eq:gblupmodel)
\end{equation}

where $W$ is the design matrix linking genomic breeding values to observations. The mixed model equations are the defined as


\begin{equation}
\left[
  \begin{array}{lr}
  X^TR^{-1}X  &  X^TR^{-1}W \\
  W^TR^{-1}X  &  W^TR^{-1}W + G^{-1}* \lambda
  \end{array}
\right]
\left[
  \begin{array}{c}
  \hat{\beta} \\
  \hat{u}
  \end{array}
\right]
=
\left[
  \begin{array}{c}
  X^TR^{-1}y \\
  W^TR^{-1}y
  \end{array}
\right]
(\#eq:gblupmme)
\end{equation}

where $G$ is defined as in \@ref(eq:genomicrelmat) and $\lambda$ is the same as in equation \@ref(eq:lambdammegpbv). Several authors have shown that both procedures (two-step and single step) are equivalent. From \@ref(eq:gblupmodel) we can see that the GBLUP model looks very similar to the animal model, except that the covariances between random effects in the animal model are based on the numerator relationship matrix and in GBLUP they are modeled via the genomic relationship matrix $G$. This means in the animal model the covariance between random breeding values is based on the concept of common ancestry and identity by descent. This is replaced in GBLUP by the concept of sharing the same alleles based on identity by state which is assumed to be the cause of the covariance between random genomic breeding values. 

The predicted genomic breeding values $\hat{u}$ coming out of \@ref(eq:gblupmme) are referred to as `direct genomic breeding values` (DGV).

### Genomic Relationship Matrix $G$
Multiplying the matrix $M$ with its transpose $M^T$ results in a $n\times n$ square matrix $MM^T$. On the diagonal of this matrix we get counts of how many alleles in each individual have a positive effect. The off-diagonal elements count how many individual share the same alleles across all SNP-positions. In contrast to the additive genetic relationship matrix $A$, the counts here are based on identity by state and not on identity by descent.

The problem with matrix $MM^T$ is its dependence on the number SNP-markers. Therefore the matrix $MM^T$ is proportional to the relationship $A$ but it does not correspond to $A$ directly. As a solution to that problem (VanRaden2008) proposed to re-scale such that allele frequencies on a given locus are expressed as to times the deviation from $0.5$. This re-scaling is done with an $n \times m$ matrix $P$ where each of the $m$ columns corresponds to a SNP-Locus. Elements in column $i$ of matrix $P$ have all the same value corresponding to $2p_i - 0.5$ where $p_i$ corresponds to the frequency of the SNP-allele associated to the positive QTL-allele at locus $i$. 

The difference between matrices $M$ and $P$ is assigned to a new matrix $Z$

$$Z = M-P$$

Finally the matrix $ZZ^T$ must be scaled with the sum of $2p_i(1-p_i)$ over all SNP-loci to get to the genomic relationship matrix $G$.

\begin{equation}
G = \frac{ZZ^T}{\sum_{i=1}^m 2p_i(1-p_i)}
(\#eq:genomicrelmat)
\end{equation}

The matrix $G$ has similar properties as the numerator relationship matrix $A$. The genomic inbreeding coefficient $F_j$ is defined as $F_j = (G)_{jj}-1$. The genomic relationship $a_{ij}$ between two individuals $i$ and $j$ corresponds to the element in matrix $G$ divided by the square root of the diagonal elements 

$$a_{ij} = \frac{G_{ij}}{\sqrt{G_{ii}G_{jj}}}$$



### Practical Problems {#practical-problems}
The model equations \@ref(eq:gblupmodel) look very straight-forward, but the practical implementation can be quite complicated. The reason for these problems is the fact that compared to the total size of a population only a small fraction of all animals are genotyped and hence contribute the the genomic evaluation. On the other hand DGV do not contain all information that occur in conventional breeding values.

Because all non-genotyped offsprings of parents are ignored by GBLUP, this loss of information is even more dramatic. For the two step-procedure as long as the reference population has a reasonable size and is not too heterogeneous, this is not a problem, we can still come up with reasonable estimates of SNP-effects. Due to the in-balanced availability of genotypic information, a procedure to combine DGV with traditional predicted breeding values was adopted. This procedure starts with predicting DGV and combining them with traditionally predicted breeding values from parents which are termed as parent averages (PA). This procedure of combining predicted breeding values from different sources is called __blending__. The problem with blending one has to be aware of is that there is a covariance between DGV and PA which must be accounted for.

A further problem is that there are different techniques to generate genotyping results. The different results also have different densities which means that they give different numbers of SNP-loci per genome. The different techniques also vary in price which is the reason that genotyping results from different technologies must be combined. Combining genotyping results with different densities of SNP-markers per genome is done with a process that is called __imputing__. This basically comes done to inferring missing SNP-genotypes on marker panels with less density based on results from denser marker panels.


### How Does GBLUP Work {#asm-gblup-howdoesgblupwork}
The genomic relationship matrix $G$ allows to predict genomic breeding values for animals with SNP-Genotypes without any observation in the dataset. This fact is the basis of the large benefit of genomic selection. As soon as a young animal is born, its SNP genotypes can be determined and a genomic breeding value can be predicted. This genomic breeding value is much more accurate then the traditional breeding value based only on ancestral information. 

The BVM model given in \@ref(eq:asmgblupbvm) is a mixed linear effects model. The solution for the unknown parameters can be obtained by solving the mixed model equations shown in \@ref(eq:asm-gblup-gblupmme). In this form the Inverse $G^{-1}$ of $G$ and the vector $\hat{g}$ of predicted genotypic breeding values are split into one part corresponding to the animals with observations and a second part for the animals without phenotypic information.

```{r GblupMme, echo=FALSE, results='asis'}
matCoeff <- matrix(c("X^TX","X^TZ","0",
                     "Z^TX","Z^TZ + G^{(11)}","G^{(12)}",
                     "0","G^{(21)}","G^{(22)}"), ncol = 3, byrow = TRUE)
vecSol <- c("\\hat{b}","\\hat{g}_1","\\hat{g}_2")
vecRhs <- c("X^Ty","Z^Ty","0")
### # show mme
cat("\\begin{equation}\n")
cat(paste(rmdhelp::bmatrix(pmat = matCoeff), collapse = '\n'), '\n')
cat(paste(rmdhelp::bcolumn_vector(pvec = vecSol), collapse = '\n'), '\n')
cat(" = \n")
cat(paste(rmdhelp::bcolumn_vector(pvec = vecRhs), collapse = '\n'), '\n')
cat("(\\#eq:asm-gblup-gblupmme)")
cat("\\end{equation}\n")
```

The matrix $G^{(11)}$ denotes the part of $G^{-1}$ corresponding to the animals with phenotypic observations. Similarly, $G^{(22)}$ stands for the part of the animals without genotypic observations. The matrices $G^{(12)}$ and $G^{(21)}$ are the parts of $G^{-1}$ which link the two groups of animals. The same partitioning holds for the vector of predicted breeding values. The vector $\hat{g}_1$ contains the predicted breeding values for the animals with observations and the vector $\hat{g}_2$ contains the predicted breeding values of all animals without phenotypic observations. 

Based on the last line of \@ref(eq:asm-gblup-gblupmme) the predicted breeding values $\hat{g}_2$ of all animals without phenotypic observations can be computed from the predicted breeding values $\hat{g}_1$ from the animals with observations.

\begin{equation}
\hat{g}_2 = -\left( G^{22}\right)^{-1}G^{21}\hat{g}_1
(\#eq:GenomicBvAnimalNoPhen)
\end{equation}

Equation \@ref(eq:GenomicBvAnimalNoPhen) is referred to as genomic regression of predicted breeding values of animals without observation on the predicted genomic breeding values of animals with observations.


## Accuracy of genome-wide EBV {} 
In the example above, EBV were not estimated very accurately because only 6 phenotypes and 
genotypes were available for only two markers. The accuracy of genomic EBV increases when the 
size of the reference population increases, when the reference population represents as much of the 
relevant genetic variation in the population as possible, when selection candidates are closely 
related to the reference population, when genetic diversity in the population is low (i.e. low 
effective population size) and with better statistical models. More informative marker maps also 
increase accuracy, although the increase here is marginal when the marker density is already high 
(i.e. 50,000 markers for within breed selection; advantageous with more markers for very 
heterogeneous populations, across-breed analyses). 

The accuracy is trait dependent and depends on heritability. Figure 2 shows how the accuracy of 
EBV depends on heritability and number of phenotypic records.

Figure 2. Accuracy of EBV (rIA) as a function of number of phenotypic records (n) used to estimate 
marker effects for traits with heritability of 0.4 (solid curve), 0.15 (dotted curve) and 
0.05 (broken curve); Assumes normal distribution of marker effects and effective population size equal to 100. 


A common finding is that a straightforward BLUP method for estimating the marker effects gave reliabilities of GEBV almost as high as more complex methods. The BLUP method is attractive because the only prior information required is the additive genetic variance of the trait. 



## Impact of genome-wide information on breeding programmes {-}
The impact of using genomic information depends on the efficiency of traditional breeding that 
does not use genomic information. If all selection candidates already have accurate EBV at the time 
of selection then genomic information will not add much, if anything. Hence, genome-wide marker 
information is most useful when phenotypic recording is restricted – for instance when phenotypes 
are expressed late in the animal’s life (e.g. meat quality, longevity), are expressed only in one sex 
(e.g. milk yield, egg production) or are expensive to measure (e.g. feed efficiency, bacteriological 
samples, progesterone profiles or other physiological measures). Furthermore, genome-wide marker 
information is useful for traits with low heritability provided a sufficient amount of phenotypes can 
be recorded. It should therefore be clear that the extra benefits of genome-wide information vary 
across traits and across species although it can in principle be used for all species and traits.

Dairy cattle breeding is characterised by the main traits only being measurable in females while 
very intense selection is only possible in males. Thus genome-wide markers are very useful in dairy 
cattle breeding. 
In pig breeding, most traits are measured on all selection candidates before sexual 
maturity. Therefore genomic information gives less extra value for pig breeding compared with 
dairy cattle. However, exceptions for pigs include litter size (only measurable in females and after 
sexual maturity), feed efficiency (only measured on few animals because it is expensive), longevity 
and carcass traits (expressed late). Another potential benefit for pigs is that traits can be recorded on 
crossbreed production animals which may be housed in different production environments and have 
different effects of single genes compared with purebred animals. 

Genomic breeding values can be used to enhance the screening of potential breeding animals for 
testing (pre-selection) which is especially attractive in situations when the costs of genotyping are 
relatively inexpensive compared to the costs of recording phenotypes. They are also useful in 
intensifying selection for young animals thereby facilitating a reduced overall generation interval. 
For instance, with the availability of accurate genome-wide breeding values for young bulls it is 
more attractive to use the best young bulls widely rather than having to wait for the results of 
progeny group testing. Here the substantial reduction in generation interval offset the slightly lower
accuracy of genome-wide breeding values compared with breeding values based on progeny results. 
If, for example, the generation interval (L) is reduced from 5 to 2 years then a reduction in accuracy
(rIA) from 90% to 70% is acceptable as the genetic progress is almost doubled (i.e. increases by a 
factor 0.7×5 / 2×0.9 = 1.94). 

Another benefit of using genome-wide EBV rather than traditional EBV is that it results in less 
inbreeding if the same selection intensities are maintained. This happens because breeding based on 
traditional EBV puts more emphasis than genome-wide EBV on parent information, especially for 
traits with low heritability. 

A potential danger with genome-wide EBV is that it does not capture the effect of new non-recurrent mutations in selection candidates without phenotypic information (relating to self or progeny). Thus if selection and mating decisions are made before there is phenotypic information available from progeny, or the animal itself, it becomes impossible to estimate the effect of a new mutation. This means that new unfavourable mutations may be easier spread in the population and that favourable mutations may be missed if selection is based entirely on genome-wide EBV with 
negative consequences for long term genetic progress. 

Systems genetics approaches using prior knowledge from same or other species/breeds may be used to incorporate information about some of the rare alleles with low frequency ignored in current genomic selection. Such prior information could also sharpen inferences of genes with intermediate effects that are not ignored in current 
genomic selection. 

Also, often some traits that should be in the breeding goal are not systematically recorded (e.g. 
many diseases). But such traits are still influenced by selection on other traits and frequently the 
combined correlated effect on such non-recorded traits is negative. With selection based on 
genome-wide EBV there is a risk that the negative pressure on non-recorded traits increases. So, 
although genome-wide breeding values offers exiting opportunities for enhancing genetic progress 
by allowing for accurate selection of young animals then they should be used with appropriate care. 

For instance, the use of young bulls could be limited (e.g. to max ~10.000 inseminations and max 
~5% of the cow population) until progeny information is available. Such limits would also have a 
large positive effect in controlling inbreeding and a smaller effect on short term genetic gain. How 
to optimally design breeding plans based on genome-wide information is an important topic where 
more research is needed. 

\begin{comment}

Summary 

 * The following 3 steps are performed to calculate genome-wide breeding values: 1) genotype 
and phenotype reference animals, 2) estimate additive effects of small chromosome 
segments by regressing genotypes of reference animals on their phenotypes, 3) the breeding 
value of an animal is estimated as the sum of the effects of all chromosome segments carried 
by the animal

 * Genome-wide breeding values can be calculated both for animals with and without 
phenotypes as long as a DNA sample of the animal can be taken (potentially at embryo 
stage)

 * Genome-wide information allows accurate selection of young animals for all traits that can 
be recorded

 * Genome-wide breeding values are especially beneficial when traditional selection is difficult 
such as when phenotypic recording is restricted by sex and age (e.g. very beneficial for dairy 
cattle)

 * Conservative use of young animals without phenotypic information (relating to self or 
progeny) is advised to avoid potential negative side effects related to unfavourable 
mutations, unfavourable selection pressure on non-recorded breeding goal traits and high 
rates of inbreeding

Nowadays the term `genomic selection` is often used ambiguously. What most people mean when they are talking about GS should better be called __genomic prediction__ of breeding values. This prediction can be done in different ways which are listed below

* Two-step procedure: Effects of SNPs are predicted using single locus models in a reference population which corresponds of mainly male breeding animals with transformed predicted traditional BLUP-breeding values with an accuracy above a certain threshold. Alternatively, it is also possible to use statistics of daughter yields as observations for the prediction of marker effects. Predictions of genomic breeding values for all animals in the population with genomic information are computed by summing up all previously estimated SNP-effects. This procedure is currently applied in the Swiss dairy cattle populations.
* Single-step procedures try to predict genomic breeding values and traditional breeding values in a single evaluation.



## Genome-wide EBV {-}
Breeding values are conceptually simple to calculate from marker information. First, the entire 
genome is divided into small chromosome segments by dense markers. Second, the additive effects 
of each chromosome segment are estimated simultaneously. Finally, the genomic EBV equals the 
sum of all chromosome segment effects. The chromosome segment effects are estimated for a group 
of animals (i.e. a reference population). For any remaining animal, only a blood or tissue sample is 
needed to determine its genome-wide (or genomic) EBV (Figure 1). For breeding purposes, it is 
desirable that the EBV can be estimated accurately early in the animals life. 
The effect of each of these small chromosome segments can be estimated if we have phenotypes 
and genotypes from a number of animals. With sufficiently dense marker maps, the chromosome 
segment effects apply to all animals in the population in which they were estimated, because 
markers are in linkage disequilibrium with the causal gene that they bracket. 

Simple example: estimation of genome-wide breeding values 
Consider the genotype results of 6 animals that have each been genotyped for 2 SNP markers (Table 
2): 

Simple example: estimation of genome-wide breeding values 
Consider the genotype results of 6 animals that have each been genotyped for 2 SNP markers (Table 
2): 
Table 2. Pedigree, nucleotides at two positions, and phenotype for 6 example animals 
\begin{center} 
\begin{tabular}{|c|c|c|c|c|c|}
  \hline
Animal & Sire & Dam & Locus 1 & Locus 2 & Gain (g/day) \\
  \hline
1 & ? & ? & AA & CT & 1943 \\ 
  \hline
2 & ? & ? & GG & CC & 1908 \\ 
  \hline
3 & 1 & 2 & AG & CT & 1853 \\ 
  \hline
4 & 1 & 2 & AG & CT & 1845 \\ 
  \hline
5 & 3 & 4 & AG & CC & 2006 \\
  \hline
6 & 3 & 4 & AA & CT & 1941 \\
  \hline
\end{tabular}
\end{center}

First we check if there are inconsistencies between the pedigree and genotypes. For instance, if a 
parent is homozygous A at one marker locus then its entire offspring must have at least one A-allele 
at the same position. There are no inconsistencies in this example data, but in large real data sets 
errors can be present. 
Now we want to estimate the effect of substituting A with G at marker locus 1 and the effect of 
substituting T with C at locus 2. Here we illustrate the estimation of ‘additive allele substitution 
effects’ by simple linear regression for the purpose of simplicity, although better methods are 
available. The following linear regression model is used: 

\begin{align}
y_i &= \mu + b_1X_{1i} + b_2X_{2i} + e_i
\end{align}

where 

\begin{align}
 y_i &= \quad \text{is the daily gain of animal i (known)} \notag \\
 \mu &= \quad \text{is the mean daily gain (unknown parameter)}  \notag \\
b_1 &= \quad \text{is the additive effect of substituting A with G at locus 1 (unknown parameter)}  \notag \\
b_2 &= \quad \text{is the additive effect of substituting T with C at locus 2 (unknown parameter)} \notag \\
X_1 &= \quad \text{equals the number of A-alleles the i’th animal carries at locus 1 (known)} \notag \\
X_2 &= \quad \text{equals the number of T-alleles the i’th animal carries at locus 2 (known)}\notag \\ 
e_i &= \quad \text{is a random residual effect; residuals are assumed independent and N(0,$\sigma^2_{e}$)}\notag  
\end{align}


Using standard regression theory, the solutions to the unknown parameters are: 
$\mu$=1909.7; $b_1$=94.7; $b_2$ = -156.2; $\sigma^2_{e}$ = 199.1.


The breeding values, estimated using marker information, are: 

\begin{align}
EBV = \mu + b_1X_{1i} + b_2X_{2i}  
\end{align}

where the estimated values of $\mu$, $b_1$ and $b_2$ are used (hence estimated and not true BV). For animals 1-6, the following EBVs are obtained: 1942.8, 1909.7, 1848.2, 1848.2, 2004.3, and 1942.83 g/day. Animals 
1 and 6 have the same EBV because they have identical marker genotypes. Animal 5 has the largest 
EBV because it has two of the very favourable C-alleles at locus 2 and one of the favourable A alleles at locus 1. 
We assumed that the two markers were able to explain all genetic variation with respect to daily 
gain (i.e. all causative genes are in linkage disequilibrium with at least one of the markers). 
Normally we need thousands of markers covering the entire genome to ensure this. To accurately 
estimate the effect of many (small) chromosome segments we need many phenotypic observations 
(much more than the 6 available ones in this example; see also Fig. 2). 
Now imagine that animals 5 and 6 are mated with each other and produces a new offspring with 
genotype AA (locus 1) and CC (locus 2). 
The EBV of this new offspring is:

\begin{align}
EBV &= \mu + b_1X_{1i} + b_2X_{2i} = 2411.3 g/day. 
\end{align}

Note that it was possible to calculate an EBV for this new offspring without having 
phenotypic information for this animal. This feature makes genomic selection very useful because it 
facilitates early selection.


Statistical model and assumptions:
\begin{align}
y &= \mu + Wb + e \notag \\
b &\sim N(0,I\sigma^2_{b}) \notag \\
e &\sim N(0,I\sigma^2_{e}) \notag \\
Var(y) &= WW'\sigma^2_{b}+ I\sigma^2_{e} \notag
\end{align}

Estimating marker effects using the BLUP method:
\begin{align}
\hat{b} &= [WW'+I\frac{\sigma^2_{b}}{\sigma^2_{e}}]^{-1}W'(y-\hat{\mu})
\end{align}

Estimating genomic breeding value:
\begin{align}
\hat{a} &= W\hat{b}
\end{align}


Statistical model and assumptions:
\begin{align}
y &= \mu + a + e \notag \\
a &\sim N(0,G\sigma^2_{a}) \notag \\
e &\sim N(0,I\sigma^2_{e}) \notag \\
Var(y) &= G\sigma^2_{a}+ I\sigma^2_{e} \notag \\
G &= \frac{1}{m}WW' \qquad \text{(G is the genomic relationship matrix)} \notag
\end{align}


Estimating genomic breeding value using the BLUP method:
\begin{align}
\hat{a} &= G[G\sigma^2_{a}+I\sigma^2_{e}]^{-1}(y-\hat{\mu})
\end{align}




```{r,echo=FALSE, results='asis'}
nobs <- 6
m <- 2
muhat <- 1909.7
sigma2_e <- 1
sigma2_a <- 0.5
sigma2_b <- sigma2_a/m
Ie <- diag(1,nobs)
Ib <- diag(1,m)
### # design matrix X
X <- matrix(1, nrow = nobs, ncol = 1)
P <- matrix(2*0.5-0.5, nrow = nobs, ncol = 2)
### # design matrix Z
M <- c(2, 0, 1, 1, 1, 2, 1, 0, 1, 1, 0, 1)
M <- matrix(M, nrow = nobs, byrow = FALSE)
y <- c(1943,1908,1853,1845,2006,1941)
y <- matrix(y, nrow = nobs, ncol = 1)
sc <- 2*0.5*(1-0.5) +2*0.5*(1-0.5)
W <- (M-P)/(2*0.5*(1-0.5))
G <- (M-P)%*%t((M-P))/sc
bhat1 <- solve(t(M)%*%M)%*%t(M)%*%(y-muhat)
ghat1 <- M%*%bhat1
bhat2 <- solve(t(W)%*%W+Ib*sigma2_e/sigma2_b)%*%t(W)%*%(y-muhat)
ghat2 <- W%*%bhat2
ghat3 <- G%*%solve(G*sigma2_a+Ie*sigma2_a)%*%(y-muhat)
cat("$$\n")
cat("y = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = y, pnDigits = 0), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("X = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = X, pnDigits = 0), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("M = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = M, pnDigits = 0), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")

cat("$$\n")
cat("P = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = P, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("M-P = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = M-P, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")

cat("$$\n")
cat("MM' = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = M%*%t(M), pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("G = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = G, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")


cat("$$\n")
cat("g1 = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = ghat1, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("g2 = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = ghat2, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("g3 = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = ghat3, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")

```

\end{comment}
