---
title: "Estimation of Genomic Breeding Values"
author: "Guillaume Ramstein & Peter Sørensen"
date: "`r Sys.Date()`"
output:
  bookdown::pdf_document2:
    citation_package: natbib
    number_sections: yes
    includes:
      in_header: preamble.tex
  html_document:
    number_sections: yes
    includes:
      in_header: mathjax_header.html
  word_document: default
bibliography: [qg2021.bib]
link-citations: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(19)
```



## Learning objective:  {-}    
This section introduces the basic concepts of estimating genomic breeding values such as: 

   * basic principle behind estimating genomic breeding values
   * accuracy of estimated genomic breeding values
   * use of genomic relationships for estimating breeding values
   * different methods, data sources and experimental designs for estimating genomic breeding values


# Introduction
A new technology called __genomic selection__ has revolutionized animal and plant breeding. Genomic selection refers to selection decisions based on genomic estimated breeding values (GEBVs). We have previously learned how phenotypic records and genetic relationships computed from pedigree information can be used to estimate breeding values (EBVs). Genome-wide DNA markers can replace or supplement pedigree information for this purpose. The first ideas of genomic prediction were presented by (Meuwissen2001a). They showed that information from genotypes of very many marker loci evenly spread over the complete genome can successfully be used to estimate genomic breeding values. Because the genomic marker data is spread over the complete genome it is often referred to as __genomic information__ and from the use of this information for selection purposes the term of __genomic selection__ was coined. The early results on genomic selection were not considered until the paper by (Schaeffer2006) showed that in a cattle breeding program the introduction of genomic selection could lead to savings of about 90% of the total costs, provided that the accuracies computed by (Meuwissen2001a) can really be achieved. After the publication of (Schaeffer2006) many animal and plant breeding organisation started to introduce procedures for genomic selection. 


# Genomic information
In recent years much attention has been given to genomic information due to the dramatic development in genotyping technologies. Today dense genetic maps are available for most of the most important animal and plant species (Table 1). It is, 
however, still lacking for several species, but this lack of genomic information can be circumvented by using RAD-sequencing (Restriction site associated DNA sequencing) or Genotyping-by-Sequencing (GBS), which enable dividing the entire genome into smaller segments. Ultimately the entire genome may be sequenced. This is possible, but still very expensive, so only a few founder individual have been fully 
sequenced (in animals, mostly bulls, but also some horses and dogs; in plants, dozens of accessions in major crops like maize, rice, barley, and wheat). Lower resolution maps are also used and especially in cattle to save costs (e.g. females). 

Table 1. Number of markers used for genomic selection in different species 
\begin{center} 
\begin{tabular}{|c|c|c|}
  \hline
  Species & No. SNPs (in thousands) & Genome size (x$10^9$) \\
  \hline
  Cattle & 778 & 2.67 \\
  \hline
  Pig & 64 & 2.81  \\
  \hline
  Chicken & 581 & 1.05 \\ 
  \hline
  Horse & 70 & 2.47  \\
  \hline
  Sheep & 54 & 2.62  \\
  \hline
  Dog & 170 & 2.41  \\
  \hline
  Maize	&	* &	2.3  \\
  \hline
  Wheat	&	*	& 17.1  \\
  \hline
  Rice	&	*	& 0.38  \\
  \hline
  Barley &		*	& 5.3  \\
  \hline
  Tomato	&	*	& 0.83  \\
  \hline
\end{tabular}
\end{center}
*: Number of SNPs vary greatly  depending on the assay: from about 6,000 to 900,000 in SNP arrays, to several millions in whole-genome sequencing data.

The genetic maps are based on DNA markers in the form of single nucleotide polymorphisms (SNP) and they enable us to divide the entire genome into thousands of relatively small chromosome segments. 
 
## Genomic markers
The different locations in the genome that are considered in genomic selection are called __markers__. When looking at the complete set of markers making up the genomic information in a population, the so-called __Single Nucleotide Polymorphisms__ (SNPs) have been shown to be the most useful types of markers. These SNPs correspond to differences of single bases at a given position in the genome. Based on empirical analyses of very many SNP-loci, almost all SNP just take two different states. Furthermore it is important that these SNPs are more or less evenly spread over the complete genome. Some SNPs may be located in coding regions and some my be placed in regions of unknown function. Figure \@ref(fig:snpdistribution) illustrates the distribution of SNP over the genome. 

```{r snpdistribution, echo=FALSE, hook_convert_odg=TRUE, fig_path="odg", fig.cap="Distribution of SNP-Loci Across A Genome"}
#rmddochelper::use_odg_graphic(ps_path = "odg/snpdistribution.odg")
knitr::include_graphics(path = "odg/snpdistribution.png")
```

## Quantitative Trait Loci and linkage disequilibrium
The loci that are relevant for a quantitative trait are called __Quantitative Trait Loci__ (QTL). Any given SNP-Marker can only be informative for a given QTL, if a certain __linkage disequilibrium__ between the QTL and the marker locus exists. The idea behind linkage disequilibrium is that a certain positive QTL-allele evolved in a certain genetic neighborhood of a number of SNP loci. As a result of that the positive QTL-allele is very often inherited with the same SNP-allele. Over the generations, recombination between the QTL and the neighboring SNP-loci can happen and thereby weaken the statistical association between the positive QTL-allele and the given SNP-allele. This recombination effect is smaller when the QTL and the SNP-loci are physically closer together on the chromosome. The non-random association between QTL and SNP-markers is called linkage disequilibrium.

The marker locus is called $M$ and the QTL is called $Q$, then the LD can be measured by 

\begin{align}
D &= p(M_1Q_1) * p(M_2Q_2) - p(M_1Q_2) * p(M_2Q_1)
(\#eq:linkagediseq)
\end{align}

where $p(M_xQ_y)$ corresponds to the frequency of the combination of marker allele $M_x$ and QTL allele $Q_y$. Very often the LD measure shown in \@ref(eq:linkagediseq) is re-scaled to the interval between $0$ and $1$ which leads to 

\begin{align}
r^2 &= \frac{D^2}{p(M_1)*p(M_2) * p(Q_1) * p(Q_2)}
(\#eq:linkagediseqrescaled)
\end{align}

In \@ref(eq:linkagediseqrescaled) $r^2$ describes the proportion of the variance at the QTL which is explained by the marker $M$. Hence the LD must be high such that the marker can explain a large part of the variance at the QTL. For the genome size of most animal and plant species, about $50,000$ SNP markers are required to get a sufficient coverage of the complete genome.

# Basic principles for estimating genomic breeding values
Genomic breeding values are conceptually simple to calculate from marker information. First, the entire genome is divided into small chromosome segments by dense markers. Second, the additive effects of each chromosome segment are estimated simultaneously. Finally, the genomic estimated breeding value (GEBV) is calculated as the sum of all chromosome segment effects. The chromosome segment effects are estimated for a group of individuals (i.e. a reference population). For any remaining individual, only a blood or tissue sample is 
needed to determine its genome-wide (or genomic) breeding values. For breeding purposes, it is desirable that the GEBV can be estimated accurately, and early in the individual's life. The effect of each of these small chromosome segments can be estimated if we have phenotypes and genotypes from many individuals (from several hundreds to hundreds of thousands or even millions). With sufficiently dense marker maps, the chromosome segment effects capture the genomic variability in the population in which they were estimated, because markers are in linkage disequilibrium with the causal gene that they bracket. 

We will present two approaches that are commonly used to estimate genomic breeding values. The first approach is referred to as MBLUP (or SNP-BLUP). In this approach marker effects are estimated from observed phenotypic and genomic marker data recorded in the reference (or training) population. Genomic breeding values are estimated from the marker effects and genomic marker data for potential selection candidates in the breeding population. 
The second approach is referred to as GBLUP. In this approach genomic breeding values are estimated from observed phenotypic and genomic marker data for individuals in the reference (or training) population. Genomic breeding values for selection candidates are estimated based on their __genomic relationship__ to individuals in the reference population.
Both approaches allow for estimation of genomic breeding values for individuals without phenotypes and close relationships. This is one of the main advantages the genomic prediction and selection. As soon as DNA is available for an individual, its marker genotypes can be determined and a genomic breeding value can be estimated. Furthermore, GEBVs are generally more accurate than the traditional estimated breeding values (EBVs) based only on pedigree information. 


## A linear mixed model for estimating marker effects (MBLUP) {#linear-model-pgbv}
The linear mixed model for estimating marker effects contains the observation vector for the trait(s) of interest ($y$), the fixed effects $b$, which explain systematic differences in $y$, the random marker effects $s$, and the random residual effects $e$. A matrix formulation of a general linear mixed model for estimating marker effects is:
\begin{align}
y &= Xb + Ms + e 
(\#eq:mblup)
\end{align}

where
\begin{align}
y &: \text{is the vector of observed values of the trait,} \notag \\
X &: \text{is a known design matrix that relates the elements of $b$ to their corresponding element in $y$.} \notag \\
b &: \text{is a vector of fixed effects,} \notag \\
M &: \text{is a known design matrix that relates the elements of $s$ to their corresponding element in $y$.} \notag \\
s &: \text{is a vector of random marker effects,} \notag \\
e &: \text{is a vector of random residual effects,} \notag 
\end{align}

In the linear mixed model above the marker and residual effects ($s$ and $e$) and the phenotypes ($y$) are considered to be random variables which follow a multivariate normal (MVN) distribution. In general terms the distributions of these random variables are:  
\begin{align}
s \sim MVN(0,S) \notag \\
e \sim MVN(0,R) \notag \\
y \sim MVN(Xb,V) \notag
\end{align}
where $S=I_s\sigma_s^2$ is a square matrix of (co)variances among marker effects (usually assumed independent), and $R=I\sigma_e^2$ is a square matrix of residual (co)variances among residuals (also assumed independent, most of the times), and $V=MM'\sigma_s^2+I\sigma_e^2$ is the overall phenotypic covariance matrix. 

The marker variance $\sigma_s^2$ is defined as:

\begin{equation}
\sigma_s^2 = \frac{\sigma_a^2}{\sum_{j=1}^m 2p_j(1-p_j)} 
(\#eq:lambdammegpbv)
\end{equation}

where $\sigma_a^2$ is the total additive genetic variance, $m$ is the number of markers, and $p_j$ is the frequency of the marker-allele that is associated with the positive QTL-allele.


### Estimation of marker effects in MBLUP
Estimates of the fixed effect $b$, and random marker effects, $s$, in the linear mixed model specified above can be done using the BLUE and BLUP equations shown below.

The best linear unbiased estimator (BLUE) of the fixed effects $\hat{b}$ is:

\begin{equation}
\hat{b} = (X'V^{-1} X)^{-1} X' V^{-1} y
(\#eq:hatbetahatblue)
\end{equation}

The best linear unbiased prediction (BLUP) of the marker effects $\hat{s}$ is:

\begin{equation}
\hat{s} = SM'V^{-1}(y - X\hat{b})
(\#eq:hatublup)
\end{equation}

The BLUP equation for the estimate of the marker effects consists of three parts; The term, $y - X\hat{b}$, shows that the observed phenotypic values are corrected for the fixed effects represented by $X\hat{b}$. The covariance between the true marker effects ($s$) and phenotypes ($y$) is $Cov(s,y)=SM'=M'\sigma_s^2$. The inverse of the phenotypic covariance matrix is $[Var(y)]^{-1}=V^{-1}$. Alternatively, estimates of the (fixed and random) effects in the model can be obtained by solving the mixed model equations. The mixed-model equations for the model given in \@ref(eq:mblup) have the following structure:

\begin{equation}
\left[
  \begin{array}{lr}
  X^TR^{-1}X  &  X^TR^{-1}M  \\
  M^TR^{-1}X  &  M^TR^{-1}M + S^{-1}
  \end{array}
\right]
\left[
  \begin{array}{c}
  \hat{b}\\
  \hat{s}
  \end{array}
\right]
=
\left[
  \begin{array}{c}
  X^TR^{-1}y \\
  M^TR^{-1}y
  \end{array}
\right]
(\#eq:mmegpbv)
\end{equation}

### Estimation of genomic breeding values  in MBLUP
The estimates of the marker effects $\hat{s}$ in \@ref(eq:mmegpbv) can be used to estimate genomic breeding values $\hat{a}$ for any individual with genomic information (i.e. genotypes for the same set of markers used in the reference population) by:

\begin{equation}
\hat{a} = \sum_{j=1}^m M_j \hat{s}_j
(\#eq:genomicpbv)
\end{equation}

where $M_j$ corresponds to the vector of observed marker genotypes of an individual. 

### Encoding of the marker genotype matrix $M$
The elements in the matrix $M$ can be encoded in different ways. The results from the genotyping laboratory represents the nucleotides found at a given genome position. To be used in the linear model the nucleotides (genotypes) at each position (marker locus) must be encoded numerically. Let us assume that at a given SNP-position, the bases $G$ or $C$ are observed and $G$ corresponds to the allele with the positive effect on our trait of interest. Based on the two observed alleles, the possible genotypes are $GG$, $GC$ or $CC$. One possible code for this SNP in the matrix $M$ might be the number of $G$-alleles which corresponds to $2$, $1$ and $0$. Alternatively, it is also possible to use the codes $1$, $0$ and $-1$ instead which corresponds to the factors with which $a$ is multiplied to get the genotypic values in the single locus model.



## A linear mixed model for estimating genomic breeding values (GBLUP) {#gblup}
The linear mixed model for estimating genomic breeding values (GBLUP) contains the observation vector for the trait(s) of interest ($y$), the fixed effects $b$ that explain systematic differences in $y$, and the random genomic effects $a$ and random residual effects $e$. A matrix formulation of a general GBLUP model is:
\begin{align}
y &= Xb + Za + e 
(\#eq:gblup)
\end{align}

where
\begin{align}
y &: \text{is the vector of observed values of the trait,} \notag \\
b &: \text{is a vector of fixed effects,} \notag \\
a &: \text{is a vector of random genomic effects,} \notag \\
e &: \text{is a vector of random residual effects,} \notag \\
X &: \text{is a known design matrix that relates the elements of $b$ to their corresponding element in $y$.} \notag \\
Z &: \text{is a known design matrix that relates the elements of $a$ to their corresponding element in $y$.} \notag 
\end{align}

In the linear mixed model (specified above) the genomic and residual effects ($a$ and $e$) and the phenotypes ($y$) are considered to be random variables which follow a multivariate normal (MVN) distribution. In general terms the distributions of these random variables are:  
\begin{align}
a \sim MVN(0,\tilde{G}) \notag \\
e \sim MVN(0,R) \notag \\
y \sim MVN(Xb,V) \notag
\end{align}
where $\tilde{G}=G\sigma_a^2$, and $R=I\sigma_e^2$ are square matrices of genomic and residual (co)variances among the individuals, respectively, and $V=G\sigma_a^2+I\sigma_e^2$ is the overall phenotypic covariance matrix. 
The genomic relationship matrix $G$ is estimated from genomic marker data instead of pedigree information. 


### Estimation of genomic effects in GBLUP
Estimates of the fixed effect $b$, and random genomic effects, $a$, in the linear mixed model specified above can be done using the BLUE and BLUP equations shown below.

The best linear unbiased estimator (BLUE) of the fixed effects $\hat{b}$ is:

\begin{equation}
\hat{b} = (X'V^{-1} X)^{-1} X' V^{-1} y
(\#eq:hatbetahatblue)
\end{equation}

The best linear unbiased prediction (BLUP) of the genomic effects $\hat{a}$ is:

\begin{equation}
\hat{a} = \tilde{G}Z'V^{-1}(y - X\hat{b})
(\#eq:hatublup)
\end{equation}

The BLUP equation for the estimate of the genomic effects consists of three parts; The term, $y - X\hat{b}$, shows that the observed phenotypic values are corrected for the fixed effects represented by $X\hat{b}$. The covariance between the true genomic effects ($a$) and phenotypes ($y$) is $Cov(a,y)=\tilde{G}Z'=GZ'\sigma_a^2$. The inverse of the phenotypic covariance matrix is $[Var(y)]^{-1}=V^{-1}$. Alternatively, estimates of the (fixed and random) effects in the model can be obtained by solving the mixed model equations which have the following structure:

\begin{equation}
\left[
  \begin{array}{lr}
  X^TR^{-1}X  &  X^TR^{-1}Z \\
  Z^TR^{-1}X  &  Z^TR^{-1}Z + \tilde{G}^{-1}
  \end{array}
\right]
\left[
  \begin{array}{c}
  \hat{b} \\
  \hat{a}
  \end{array}
\right]
=
\left[
  \begin{array}{c}
  X^TR^{-1}y \\
  Z^TR^{-1}y
  \end{array}
\right]
(\#eq:gblupmme)
\end{equation}

From \@ref(eq:hatublup) we can see that the GBLUP estimation procedure looks very similar to the pedigree based breeding value estimation procedure (PBLUP). In GLUP the covariances between breeding values is based on the  genomic relationship matrix $G$ which is computed from genomic markers whereas in PBLUP it is based on the numerator relationship matrix $A$ computed from pedigree information. 


### Estimation of genomic breeding values for individuals without phenotypes in GBLUP
The estimates of the genomic breeding values $\hat{a}_1$ for individuals with phenotypes can be used to estimate genomic breeding values $\hat{a}_2$ for individuals with only genomic information. Under the assumption of multivariate normality for the true genomic breeding values ($a \sim MVN(0,G\sigma_a^2)$), the expected value of the estimated genomic breeding value for individuals without phenotypes ($a_2$) conditional on the genomic breeding values for individuals with phenotypes $\hat{a}_1$ can be written as:

\begin{equation}
\hat{a}_{2} = \tilde{G}_{12}\tilde{G}_{11}^{-1}\hat{a}_{1}
(\#eq:hatublupvt)
\end{equation}

The equation given in \@ref(eq:hatublupvt) consists of three parts; The term, $\hat{a}_{1}$, represent the genomic breeding values for individuals with phenotypes in the reference population. The covariance between the true breeding values for invididuals without phenotypes ($a_2$) and individuals with phenotypes ($a_1$) is $Cov(a_1,a_2)=\tilde{G}_{12}=G_{12}\sigma_a^2$. The inverse of the genomic covariance matrix for individuals with phenotypes is $\tilde{G}_{11}^{-1}=\sigma_a^{-2}G_{11}^{-1}$. 


### Genomic Relationship Matrix $G$

The additive genomic relationship matrix $G$ is constructed using all genomic markers as follows: 

\begin{equation}
G = \frac{WW^T}{\sum_{i=1}^m 2p_i(1-p_i)}
(\#eq:genomicrelmat)
\end{equation}

where $W$ is the centered and scaled genotype matrix, and m is the total number of markers. Each column vector of $W$ was calculated as follows: $w_i=M_i-2p_i-0.5$, where $p_i$ is the minor allele frequency of the i'th genomic marker and $M_i$ is the i'th column vector of the allele count matrix, $M$, which contains the genotypes coded as 0, 1 or 2 counting the number of minor allele. The centering of the allele counts and scaling factor $\sum_{i=1}^m 2p_i(1-p_i)$ ensures that the genomic relationship matrix $G$ has similar properties as the numerator relationship matrix $A$. 

The main difference between the two types of genetic relationship matrices ($A$ and $G$) is that $A$ is based on the concept of identity by descent (sharing of the same alleles, transmitted from common ancestors) whereas $G$ is based on the concept of identity by state (sharing of the same alleles, regardless of their origin). 

\begin{comment}
## Accuracy of genomic breeding values 
Genomic breeding value is generally more accurate then the traditional breeding value based only on pedigree information. One of the reasons for this is that the genomic relationship matrix more efficient use of phenotypic information for all individuals (based on degree of allele sharing) in the estimation procedure. 
The accuracy of genomic breeding values is trait specific and depends on the heritability and the number of phenotypic records. In general the accuracy of genomic breeding values increases when the size of the reference population increases, when the reference population represents as much of the relevant genetic variation in the population as possible, when selection candidates are closely related to the reference population, when genetic diversity in the population is low (i.e. low effective population size) and with better statistical models. 

A common finding is that a straightforward BLUP method for estimating the marker effects gave reliabilities of genomic breeding values almost as high as more complex methods. The BLUP method is attractive because the only prior information required is the additive genetic variance of the trait. 

More informative marker maps also increase accuracy, although the increase here is marginal when the marker density is already high (i.e. 50,000 markers for within breed selection in dairy cattle; advantageous with more markers for very heterogeneous populations, across-breed analyses). 


## Implementation of genomic prediction in practical breeding {#practical-problems}
The model equations \@ref(eq:gblupmme) look very straight-forward, but the practical implementation can be quite complicated. The reason for these problems is the fact that compared to the total size of a population only a small fraction of all individuals are genotyped and used in the estimation of genomic breeding values. 

Because all non-genotyped offsprings of parents are ignored by GBLUP, this loss of information is even more dramatic. As long as the reference population has a reasonable size and is not too heterogeneous, this is not a problem, we can still come up with reasonable estimates of marker effects and genomic breeding values for individuals without phenotypes. Due to the in-balanced availability of genotypic information, a procedure to combine genomic breeding values with pedigree based breeding values was adopted. This procedure of combining estimated breeding values from different sources is called __blending__. 

A further problem is that there are different techniques to generate genotyping results. The different results also have different densities which means that they give different numbers of SNP-loci per genome. The different techniques also vary in price which is the reason that genotyping results from different technologies must be combined. Combining genotyping results with different densities of SNP-markers per genome is done with a process that is called __imputing__. This basically comes done to inferring missing SNP-genotypes on marker panels with less density based on results from denser marker panels.
\end{comment}


# Impact of genomic prediction and selection on breeding programmes
The relative benefit of using genomic information depends on the efficiency of traditional breeding that does not use genomic information. If all selection candidates already have accurate EBVs at the time of selection then genomic information will not add much, if anything. Hence, genome-wide marker information is most useful when phenotypic recording is restricted – for instance when phenotypes are expressed late in the animal’s life (e.g. meat quality, longevity), are expressed only in one sex 
(e.g., milk yield, egg production in animals; grain yield in plants) or are expensive to measure (e.g., feed efficiency, bacteriological samples, progesterone profiles or other physiological measures in animals; metabolic and physiologic measures in plants). Furthermore, genome-wide marker information is useful for traits with low heritability, provided a sufficient amount of phenotypes can be recorded. It should therefore be clear that the extra benefits of genomic information vary across traits and across species although it can in principle be used for all species and traits.

Dairy cattle breeding is characterised by the main traits only being measurable in females while very intense selection is only possible in males. Thus genome-wide markers are very useful in dairy cattle breeding. 

In pig breeding, most traits are measured on all selection candidates before sexual maturity. Therefore genomic information gives less extra value for pig breeding compared with dairy cattle. However, exceptions for pigs include litter size (only measurable in females and after sexual maturity), feed efficiency (only measured on few animals because it is expensive), longevity, and carcass traits (expressed late). Another potential benefit for pigs is that traits can be recorded on crossbreed production animals which may be housed in different production environments and have different effects of single genes compared with purebred animals. 

Genomic breeding values can be used to enhance the screening of potential breeding individuals for testing (pre-selection) which is especially attractive in situations when the costs of genotyping are relatively inexpensive compared to the costs of recording phenotypes. 

Genomic breeding values are also useful in intensifying selection for young animals thereby facilitating a reduced overall generation interval. For instance, with the availability of accurate genome-wide breeding values for young bulls it is more attractive to use the best young bulls widely rather than having to wait for the results of progeny group testing. Here the substantial reduction in generation interval offset the slightly lower accuracy of GEBVs compared with EBVs based on progeny results. 

Another benefit of using GEBVs rather than traditional pedigree based EBVs is that it results in less inbreeding if the same selection intensities are maintained. This happens because breeding based on traditional pedigree based breeding values puts more emphasis than genomic breeding values on parent information, especially for traits with low heritability. 

\begin{comment}
A potential danger with genomic breeding values is that it does not capture the effect of new non-recurrent mutations in selection candidates without phenotypic information (relating to self or progeny). Thus if selection and mating decisions are made before there is phenotypic information available from progeny, or the individual itself, it becomes impossible to estimate the effect of a new mutation. This means that new unfavourable mutations may be easier spread in the population and that favourable mutations may be missed if selection is based entirely on genomic breeding values with 
negative consequences for long term genetic progress. 

Also, often some traits that should be in the breeding goal are not systematically recorded (e.g. many diseases). But such traits are still influenced by selection on other traits and frequently the combined correlated effect on such non-recorded traits is negative. With selection based on genomic breeding values there is a risk that the negative pressure on non-recorded traits increases. So, although genomic breeding values offers exiting opportunities for enhancing genetic progress by allowing for accurate selection of individuals then they should be used with appropriate care. 
\end{comment}


\begin{comment}

Summary 

 * The following 3 steps are performed to calculate genome-wide breeding values: 1) genotype 
and phenotype reference animals, 2) estimate additive effects of small chromosome 
segments by regressing genotypes of reference animals on their phenotypes, 3) the breeding 
value of an animal is estimated as the sum of the effects of all chromosome segments carried 
by the animal

 * Genome-wide breeding values can be calculated both for animals with and without 
phenotypes as long as a DNA sample of the animal can be taken (potentially at embryo 
stage)

 * Genome-wide information allows accurate selection of young animals for all traits that can 
be recorded

 * Genome-wide breeding values are especially beneficial when traditional selection is difficult 
such as when phenotypic recording is restricted by sex and age (e.g. very beneficial for dairy 
cattle)

 * Conservative use of young animals without phenotypic information (relating to self or 
progeny) is advised to avoid potential negative side effects related to unfavourable 
mutations, unfavourable selection pressure on non-recorded breeding goal traits and high 
rates of inbreeding

Nowadays the term `genomic selection` is often used ambiguously. What most people mean when they are talking about GS should better be called __genomic prediction__ of breeding values. This prediction can be done in different ways which are listed below

* Two-step procedure: Effects of SNPs are predicted using single locus models in a reference population which corresponds of mainly male breeding animals with transformed predicted traditional BLUP-breeding values with an accuracy above a certain threshold. Alternatively, it is also possible to use statistics of daughter yields as observations for the prediction of marker effects. Predictions of genomic breeding values for all animals in the population with genomic information are computed by summing up all previously estimated SNP-effects. This procedure is currently applied in the Swiss dairy cattle populations.
* Single-step procedures try to predict genomic breeding values and traditional breeding values in a single evaluation.



## Genome-wide EBV {-}
Breeding values are conceptually simple to calculate from marker information. First, the entire 
genome is divided into small chromosome segments by dense markers. Second, the additive effects 
of each chromosome segment are estimated simultaneously. Finally, the genomic EBV equals the 
sum of all chromosome segment effects. The chromosome segment effects are estimated for a group 
of animals (i.e. a reference population). For any remaining animal, only a blood or tissue sample is 
needed to determine its genome-wide (or genomic) EBV (Figure 1). For breeding purposes, it is 
desirable that the EBV can be estimated accurately early in the animals life. 
The effect of each of these small chromosome segments can be estimated if we have phenotypes 
and genotypes from a number of animals. With sufficiently dense marker maps, the chromosome 
segment effects apply to all animals in the population in which they were estimated, because 
markers are in linkage disequilibrium with the causal gene that they bracket. 

Simple example: estimation of genome-wide breeding values 
Consider the genotype results of 6 animals that have each been genotyped for 2 SNP markers (Table 
2): 

Simple example: estimation of genome-wide breeding values 
Consider the genotype results of 6 animals that have each been genotyped for 2 SNP markers (Table 
2): 
Table 2. Pedigree, nucleotides at two positions, and phenotype for 6 example animals 
\begin{center} 
\begin{tabular}{|c|c|c|c|c|c|}
  \hline
Animal & Sire & Dam & Locus 1 & Locus 2 & Gain (g/day) \\
  \hline
1 & ? & ? & AA & CT & 1943 \\ 
  \hline
2 & ? & ? & GG & CC & 1908 \\ 
  \hline
3 & 1 & 2 & AG & CT & 1853 \\ 
  \hline
4 & 1 & 2 & AG & CT & 1845 \\ 
  \hline
5 & 3 & 4 & AG & CC & 2006 \\
  \hline
6 & 3 & 4 & AA & CT & 1941 \\
  \hline
\end{tabular}
\end{center}

First we check if there are inconsistencies between the pedigree and genotypes. For instance, if a 
parent is homozygous A at one marker locus then its entire offspring must have at least one A-allele 
at the same position. There are no inconsistencies in this example data, but in large real data sets 
errors can be present. 
Now we want to estimate the effect of substituting A with G at marker locus 1 and the effect of 
substituting T with C at locus 2. Here we illustrate the estimation of ‘additive allele substitution 
effects’ by simple linear regression for the purpose of simplicity, although better methods are 
available. The following linear regression model is used: 

\begin{align}
y_i &= \mu + b_1X_{1i} + b_2X_{2i} + e_i
\end{align}

where 

\begin{align}
 y_i &= \quad \text{is the daily gain of animal i (known)} \notag \\
 \mu &= \quad \text{is the mean daily gain (unknown parameter)}  \notag \\
b_1 &= \quad \text{is the additive effect of substituting A with G at locus 1 (unknown parameter)}  \notag \\
b_2 &= \quad \text{is the additive effect of substituting T with C at locus 2 (unknown parameter)} \notag \\
X_1 &= \quad \text{equals the number of A-alleles the i’th animal carries at locus 1 (known)} \notag \\
X_2 &= \quad \text{equals the number of T-alleles the i’th animal carries at locus 2 (known)}\notag \\ 
e_i &= \quad \text{is a random residual effect; residuals are assumed independent and N(0,$\sigma^2_{e}$)}\notag  
\end{align}


Using standard regression theory, the solutions to the unknown parameters are: 
$\mu$=1909.7; $b_1$=94.7; $b_2$ = -156.2; $\sigma^2_{e}$ = 199.1.


The breeding values, estimated using marker information, are: 

\begin{align}
EBV = \mu + b_1X_{1i} + b_2X_{2i}  
\end{align}

where the estimated values of $\mu$, $b_1$ and $b_2$ are used (hence estimated and not true BV). For animals 1-6, the following EBVs are obtained: 1942.8, 1909.7, 1848.2, 1848.2, 2004.3, and 1942.83 g/day. Animals 
1 and 6 have the same EBV because they have identical marker genotypes. Animal 5 has the largest 
EBV because it has two of the very favourable C-alleles at locus 2 and one of the favourable A alleles at locus 1. 
We assumed that the two markers were able to explain all genetic variation with respect to daily 
gain (i.e. all causative genes are in linkage disequilibrium with at least one of the markers). 
Normally we need thousands of markers covering the entire genome to ensure this. To accurately 
estimate the effect of many (small) chromosome segments we need many phenotypic observations 
(much more than the 6 available ones in this example; see also Fig. 2). 
Now imagine that animals 5 and 6 are mated with each other and produces a new offspring with 
genotype AA (locus 1) and CC (locus 2). 
The EBV of this new offspring is:

\begin{align}
EBV &= \mu + b_1X_{1i} + b_2X_{2i} = 2411.3 g/day. 
\end{align}

Note that it was possible to calculate an EBV for this new offspring without having 
phenotypic information for this animal. This feature makes genomic selection very useful because it 
facilitates early selection.


Statistical model and assumptions:
\begin{align}
y &= \mu + Wb + e \notag \\
b &\sim N(0,I\sigma^2_{b}) \notag \\
e &\sim N(0,I\sigma^2_{e}) \notag \\
Var(y) &= WW'\sigma^2_{b}+ I\sigma^2_{e} \notag
\end{align}

Estimating marker effects using the BLUP method:
\begin{align}
\hat{b} &= [WW'+I\frac{\sigma^2_{b}}{\sigma^2_{e}}]^{-1}W'(y-\hat{\mu})
\end{align}

Estimating genomic breeding value:
\begin{align}
\hat{a} &= W\hat{b}
\end{align}


Statistical model and assumptions:
\begin{align}
y &= \mu + a + e \notag \\
a &\sim N(0,G\sigma^2_{a}) \notag \\
e &\sim N(0,I\sigma^2_{e}) \notag \\
Var(y) &= G\sigma^2_{a}+ I\sigma^2_{e} \notag \\
G &= \frac{1}{m}WW' \qquad \text{(G is the genomic relationship matrix)} \notag
\end{align}


Estimating genomic breeding value using the BLUP method:
\begin{align}
\hat{a} &= G[G\sigma^2_{a}+I\sigma^2_{e}]^{-1}(y-\hat{\mu})
\end{align}




```{r,echo=FALSE, results='asis'}
nobs <- 6
m <- 2
muhat <- 1909.7
sigma2_e <- 1
sigma2_a <- 0.5
sigma2_b <- sigma2_a/m
Ie <- diag(1,nobs)
Ib <- diag(1,m)
### # design matrix X
X <- matrix(1, nrow = nobs, ncol = 1)
P <- matrix(2*0.5-0.5, nrow = nobs, ncol = 2)
### # design matrix Z
M <- c(2, 0, 1, 1, 1, 2, 1, 0, 1, 1, 0, 1)
M <- matrix(M, nrow = nobs, byrow = FALSE)
y <- c(1943,1908,1853,1845,2006,1941)
y <- matrix(y, nrow = nobs, ncol = 1)
sc <- 2*0.5*(1-0.5) +2*0.5*(1-0.5)
W <- (M-P)/(2*0.5*(1-0.5))
G <- (M-P)%*%t((M-P))/sc
bhat1 <- solve(t(M)%*%M)%*%t(M)%*%(y-muhat)
ghat1 <- M%*%bhat1
bhat2 <- solve(t(W)%*%W+Ib*sigma2_e/sigma2_b)%*%t(W)%*%(y-muhat)
ghat2 <- W%*%bhat2
ghat3 <- G%*%solve(G*sigma2_a+Ie*sigma2_a)%*%(y-muhat)
cat("$$\n")
cat("y = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = y, pnDigits = 0), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("X = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = X, pnDigits = 0), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("M = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = M, pnDigits = 0), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")

cat("$$\n")
cat("P = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = P, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("M-P = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = M-P, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")

cat("$$\n")
cat("MM' = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = M%*%t(M), pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("G = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = G, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")


cat("$$\n")
cat("g1 = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = ghat1, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("g2 = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = ghat2, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right] \\text{, }")
cat("g3 = \\left[\n")
cat(paste(rmddochelper::sConvertMatrixToLaTexArray(pmatAMatrix = ghat3, pnDigits = 2), collapse = "\n"), "\n")
cat("\\right]")
cat("$$\n")

```

\end{comment}

\begin{comment}
## How Does GBLUP Work?
The genomic relationship matrix $G$ allows to predict genomic breeding values for individuals with marker genotypes without any observation in the dataset. This fact is the basis of the large benefit of genomic selection. As soon as a DNA is available for an individual, its marker genotypes can be determined and a genomic breeding value can be predicted. This genomic breeding value is generally more accurate then the traditional breeding value based only on pedigree information. 

The solution for the unknown parameters can be obtained by solving the mixed model equations shown in \@ref(eq:asm-gblup-gblupmme). In this form the Inverse $G^{-1}$ of $G$ and the vector $\hat{g}$ of predicted genotypic breeding values are split into one part corresponding to the individuals with observations and a second part for the individuals without phenotypic information.

```{r GblupMme, echo=FALSE, results='asis'}
matCoeff <- matrix(c("X^TX","X^TZ","0",
                     "Z^TX","Z^TZ + G^{(11)}","G^{(12)}",
                     "0","G^{(21)}","G^{(22)}"), ncol = 3, byrow = TRUE)
vecSol <- c("\\hat{b}","\\hat{a}_1","\\hat{a}_2")
vecRhs <- c("X^Ty","Z^Ty","0")
### # show mme
cat("\\begin{equation}\n")
cat(paste(rmdhelp::bmatrix(pmat = matCoeff), collapse = '\n'), '\n')
cat(paste(rmdhelp::bcolumn_vector(pvec = vecSol), collapse = '\n'), '\n')
cat(" = \n")
cat(paste(rmdhelp::bcolumn_vector(pvec = vecRhs), collapse = '\n'), '\n')
cat("(\\#eq:asm-gblup-gblupmme)")
cat("\\end{equation}\n")
```

The matrix $G^{11}$ denotes the part of $G^{-1}$ corresponding to the individuals with phenotypic observations. Similarly, $G^{22}$ stands for the part of the individuals without genotypic observations. The matrices $G^{12}$ and $G^{21}$ are the parts of $G^{-1}$ which link the two groups of individuals. The same partitioning holds for the vector of predicted breeding values. The vector $\hat{a}_1$ contains the predicted breeding values for the individuals with observations and the vector $\hat{a}_2$ contains the predicted breeding values of all individuals without phenotypic observations. 

Based on the last line of \@ref(eq:asm-gblup-gblupmme) the predicted breeding values $\hat{g}_2$ of all individuals without phenotypic observations can be computed from the predicted breeding values $\hat{g}_1$ from the individuals with observations.

\begin{equation}
\hat{a}_2 = \left( G^{22}\right)^{-1}G^{21}\hat{a}_1
(\#eq:GenomicBvAnimalNoPhen)
\end{equation}

Equation \@ref(eq:GenomicBvAnimalNoPhen) is referred to as genomic regression of predicted breeding values of individuals without observation on the predicted genomic breeding values of individuals with observations.

\end{comment}
